File: convert_indent.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'backend', 'src', 'services', 'bookingService.ts');

let content = fs.readFileSync(filePath, 'utf-8');

// Convert each group of 2 leading spaces to a tab
const lines = content.split('\n');
const convertedLines = lines.map(line => {
  const match = line.match(/^( *)/);
  if (match) {
    const spaces = match[1];
    const numSpaces = spaces.length;
    const tabs = '\t'.repeat(Math.floor(numSpaces / 2));
    const remainder = ' '.repeat(numSpaces % 2);
    return tabs + remainder + line.substring(numSpaces);
  }
  return line;
});

const convertedContent = convertedLines.join('\n');

fs.writeFileSync(filePath, convertedContent, 'utf-8');

console.log('Conversion complete');

File: convert_indent.py
import re

file_path = r'backend\src\services\bookingService.ts'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

def convert_line(line):
    match = re.match(r'^( *)', line)
    if match:
        spaces = match.group(1)
        num_spaces = len(spaces)
        tabs = '\t' * (num_spaces // 2)
        remainder = ' ' * (num_spaces % 2)
        return tabs + remainder + line[num_spaces:]
    return line

lines = content.split('\n')
converted_lines = [convert_line(line) for line in lines]
converted_content = '\n'.join(converted_lines)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(converted_content)

print('Conversion complete')
